src/core/api/replication_execution.go

```diff
@@ -45,6 +47,46 @@ func (r *ReplicationOperationAPI) Prepare() {
 		r.SendForbiddenError(errors.New(r.SecurityCtx.GetUsername()))
 		return
 	}
+
+	// check the existence of execution if execution ID is provided in the request path
+	executionIDStr := r.GetStringFromPath(":id")
+	if len(executionIDStr) > 0 {
+		executionID, err := strconv.ParseInt(executionIDStr, 10, 64)
+		if err != nil || executionID <= 0 {
+			r.SendBadRequestError(fmt.Errorf("invalid execution ID: %s", executionIDStr))
+			return
+		}
+		execution, err := replication.OperationCtl.GetExecution(executionID)
+		if err != nil {
+			r.SendInternalServerError(fmt.Errorf("failed to get execution %d: %v", executionID, err))
+			return
+		}
+		if execution == nil {
+			r.SendNotFoundError(fmt.Errorf("execution %d not found", executionID))
+			return
+		}
+		r.execution = execution
+
+		// check the existence of task if task ID is provided in the request path
+		taskIDStr := r.GetStringFromPath(":tid")
+		if len(taskIDStr) > 0 {
+			taskID, err := strconv.ParseInt(taskIDStr, 10, 64)
+			if err != nil || taskID <= 0 {
+				r.SendBadRequestError(fmt.Errorf("invalid task ID: %s", taskIDStr))
+				return
+			}
+			task, err := replication.OperationCtl.GetTask(taskID)
+			if err != nil {
+				r.SendInternalServerError(fmt.Errorf("failed to get task %d: %v", taskID, err))
+				return
+			}
+			if task == nil || task.ExecutionID != executionID {
```

Comment:
the RC of deletion a non-existing objectd should be 200 as deletion is idempotent